signaler/signaler.go

```diff
@@ -54,3 +60,13 @@ func (s *Signaler) Signal() {
 		}
 	}
 }
+
+// Close closes the Signaler. Any channel that wasn't explicitely unregistered,
+// is closed.
+func (s *Signaler) Close() {
```

Comment:
A `Close` was missing to, just in case, close all existing channels and let any remaining clients now that nothing more will be pushed.